apps/emqx_dashboard/src/emqx_dashboard_api.erl

```diff
@@ -238,7 +291,13 @@ login(post, #{body := Params}) ->
                 })};
         {error, R} ->
             ?SLOG(info, #{msg => "dashboard_login_failed", username => Username, reason => R}),
-            {401, ?BAD_USERNAME_OR_PWD, <<"Auth failed">>}
+            %% check if this error was created by emqx_dashboard_mfa
+            case emqx_dashboard_mfa:is_mfa_error(R) of
+                true ->
+                    {403, ?BAD_MFA_TOKEN, R};
```

Comment:
Nit: bad MFA errors seem to be either missing token or bad token.  It would seem like the response status code for both would be 401 (Unauthorized), since authentication didn't complete, as opposed to 403 (Forbidden; authenticated but denied access to resource/action)?

Reply:
changed to use 401